### PR TITLE
Add 'Fake App' tokens

### PR DESCRIPTION
**Pull request by wleightond**
## Proposed changes
This PR adds Fake App tokens to Canarytokens.

Changes:
- Add frontend UI to generate tokens (including an icon selection component)
- Add new token type to backend models
- Add PWA with installation instructions for iOS and Android
- Modify switchboard to handle token triggers and extract location data if available
- Add extracted device location data (and corresponding google/apple maps links) to notification email

## Types of changes
- [x] New feature (non-breaking change which adds functionality)
- [x] Documentation Update

## Checklist
- [x] Lint and unit tests pass locally with my changes (if applicable)
- [x] I have run pre-commit (`pre-commit` in the repo)
- [x] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: .github/workflows/build_docker.yml [hunk 1]
```diff
@@ -77,12 +77,11 @@ jobs:
         npm run build
 
     - name: Commit report
-      if: steps.cache-frontend-dist.outputs.cache-hit != 'true'
```
Comment (thinkst-nick) [inline .github/workflows/build_docker.yml:80]: Do we still want this change?

File: frontend_vue/src/components/ui/IncidentDetails.vue [hunk 2]
```diff
@@ -81,7 +81,7 @@
                                 !isObject(deepestnested_val) &&
                                 isNotEmpty(deepestnested_val)
                               "
-                              :label="deepnested_key"
+                              :label="deepernested_key"
```
Comment (thinkst-nick) [inline frontend_vue/src/components/ui/IncidentDetails.vue:84]: Does this not affect other tokens?

File: frontend_vue/src/utils/tokenServices.ts [hunk 3]
```diff
@@ -367,4 +367,17 @@ export const tokenServices: TokenServicesType = {
       TOKEN_CATEGORY.MICROSOFT,
     ],
   },
+  [TOKENS_TYPE.PWA]: {
+    label: 'Fake App',
+    description: 'Get an alert when someone opens a fake app on your device.',
+    documentationLink: 'https://docs.canarytokens.org/guide/http-token.html',
```
Comment (thinkst-nick) [inline frontend_vue/src/utils/tokenServices.ts:373]: This needs to be updated

File: templates/pwa_sw.js [hunk 4]
```diff
@@ -0,0 +1,43 @@
+const CACHE_NAME = `pwa-token-v0.9`;
```
Comment (thinkst-nick) [inline templates/pwa_sw.js:1]: Either we should remove the comments, or uncomment them

**Review by thinkst-nick — CHANGES_REQUESTED** [review]

File: .github/workflows/build_docker.yml (hunk 1 above)
Comment (wleightond) [reply .github/workflows/build_docker.yml:80]: We do, the cache recovery is equivalent to building the dist, not building and committing; 
if we revert a change we still want it to build, but it doesn't if we have this check in.

File: frontend_vue/src/components/ui/IncidentDetails.vue (hunk 2 above)
Comment (wleightond) [reply frontend_vue/src/components/ui/IncidentDetails.vue:84]: It likely does, so I ran through it with SRK to double check my thinking. 
The issue here is that the data structure we're trying to render is 
```
{
    'coords': {
        'accuracy': 1,
        'latitude': 0,
        'longitude': 0,
        ...
    },
    'timestamp': 0
}
```
and this code is essentially trying to parse that nested structure and output the labels. Without this change we get `coords` as the label for latitude, longitude, accuracy, etc.:

Screenshot 2024-07-31 at 11 08 56


On reflection we should check redirect tokens in particular to see if they're being passed an idiosyncratic datastructure... I remember something about lists of values passed from the browser, so it's not just dictionaries all the way down.

Comment (wleightond) [reply frontend_vue/src/components/ui/IncidentDetails.vue]: This has been fixed :)

**Pull request merged**